bindings/rust/s2n-tls/src/testing.rs

```diff
@@ -325,3 +335,205 @@ pub fn poll_tls_pair_result(pair: &mut Pair<Harness, Harness>) -> Result<()> {
+// passed into the s2n-tls connection io contexts.
+#[allow(dead_code)]
+pub struct TestPair {
+    pub server: connection::Connection,
+    pub client: connection::Connection,
+
+    // Pin: since we are dereferencing this pointer (because it is passed as the send/recv ctx)
+    // we need to ensure that the pointer remains in the same place
+    // Box: A Vec (or VecDeque) may be moved or reallocated, so we need another layer of
+    // indirection to have a stable (pinned) reference
+    /// a data buffer that the server writes to and the client reads from
+    server_tx_stream: Pin<Box<LocalDataBuffer>>,
+    /// a data buffer that the client writes to and the server reads from
+    client_tx_stream: Pin<Box<LocalDataBuffer>>,
+}
+
+impl TestPair {
+    /// utility method to test simple handshakes
+    ///
+    /// Create a client and server from the associated `config`, and try to complete
+    /// a TLS handshake. The result of the handshake is returned.
+    pub fn handshake_with_config(config: &config::Config) -> Result<(), error::Error> {
+        Self::from_configs(config, config).handshake()
+    }
+
+    /// create a pair from a config
+    ///
+    /// A server and client connection will be created, and both will be associated
+    /// with `config`. The connections will be setup for IO over shared memory,
+    /// but no IO is performed. To handshake the connections, call `handshake()`.
+    pub fn from_config(config: &config::Config) -> Self {
+        Self::from_configs(config, config)
+    }
+
+    pub fn from_configs(client_config: &config::Config, server_config: &config::Config) -> Self {
+        let client_tx_stream = Box::pin(Default::default());
+        let server_tx_stream = Box::pin(Default::default());
+
+        let client = Self::register_connection(
+            enums::Mode::Client,
+            client_config,
+            &client_tx_stream,
+            &server_tx_stream,
+        )
+        .unwrap();
+
+        let server = Self::register_connection(
+            enums::Mode::Server,
+            server_config,
+            &server_tx_stream,
+            &client_tx_stream,
+        )
+        .unwrap();
+
+        Self {
+            server,
+            client,
+            server_tx_stream,
+            client_tx_stream,
+        }
+    }
+
+    /// create a connection ready for harness IO
+    ///
+    /// This mostly consists of setting the IO callbacks and the IO contexts, but
+    /// we also turn off blinding to avoid really slow unit tests.
+    fn register_connection(
+        mode: enums::Mode,
+        config: &config::Config,
+        send_ctx: &Pin<Box<LocalDataBuffer>>,
+        recv_ctx: &Pin<Box<LocalDataBuffer>>,
+    ) -> Result<connection::Connection, error::Error> {
+        let mut conn = connection::Connection::new(mode);
+        conn.set_config(config.clone())?
+            .set_blinding(Blinding::SelfService)?
```

Comment:
I updated the comment to be more explicit that this doesn't "turn off" blinding.

I can completely remove the self-service part if you'd prefer, but note that that will expand the scope of the refactor since `set_max_blinding_delay` is a config setting rather than a connection setting, so I'll need to change the config creation code in any test that's hitting a blinding path.

Reply:
A comment is probably sufficient. I just don't want users to be surprised, since "self-service blinding disables blinding in tests" is a useful but inaccurate simplification.